plugins/eslint-plugin-liferay/docs/rules/group-imports.md

```diff
@@ -0,0 +1,48 @@
+# Group `import` statements and `require` calls (group-imports)
+
+This rule enforces (and autofixes) that `import` statements and `require` calls &mdash; henceforth referred to as just "imports" &mdash; appear in the groups described in [our guidelines](https://github.com/liferay/liferay-frontend-guidelines/issues/60).
+
+## Rule Details
+
+Imports appear in the following groups, separated by a blank line:
+
+1. "External" imports (ie. NodeJS built-in modules and dependencies specified in "package.json"; these do not start with "." or "..").
+2. "Internal" imports (ie. local to the projects, starting with "." or "..");
+3. "Side-effect-only" imports.
```

Comment:
How can we tell the group `3` apart from the `1`? (I mean solely based on the name of the module it imports)

Reply:
```javascript
import x from 'x';

import 'boom': // <-- this one is only for side effects
```

Note that `'x'` might have side effects too, but we can't know that statically; all we can know is that `'boom'` is required *only* for its side effects. In generally anything that is not pure and idempotent is a smell, but I don't think we can make a lint against them (or at least not an `'error'`-level lint) because you sometimes do need them (eg. a module that sets up polyfills, for example).